### DISCUSSION TITLE
MultiApps

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#23527] [Q&A General]
I am running a simulation that involves a dynamical equation (magnetic Landau-Lifshitz-Gilbert equation) and an auxiliary Poisson equation to solve for the magnetostatic potential. As the time scale for the potential can be different (longer) than that of the dynamical equation, I am using a MultiApp to solve for the Poisson equation. It works, but I noticed some quirks: (1) I have to declare the potential (phi) a nonlinear variable in the parent - I cannot declare it an AuxVariable in the parent because it has to be a nonlinear variable in the MultiApp and it seems like transferring a nonlinear variable in the parent to an Aux in the MultiApp does not work; (2) as a consequence I have to define phi a nonlinear variable in the parent . But because phi lives in more blocks than the dynamical variables, I have to define Kernels on those blocks in the parent. Because there is nothing to be done on phi in the parent, I use NullKernel; (3) it appears that the code crashes (PETSc errors - matrix blocks are not defined or zero) unless I define a boundary condition on phi in the parent. Because I don't do anything on phi I just wrote a NullBC that does nothing (similar to NullKernel) but now it all works. 

Am I missing something here? It all would seem a lot simpler if I could transfer the nonlinear variable phi to an Aux variable in the MultiApp.

Thanks,
Olle

**Comment by GiudGiud** [comment]
Hi Olle

We can transfer nonlinear variables to auxvariables, and even in the other direction to initialize nonlinear variables before solving.
Which transfer are you using?

Guillaume

**Reply by heinono1** [reply]
Hi Guillaume. I am using MultiAppCopyTransfer.

**Reply by GiudGiud** [reply]
ok this might be gap in the capability. I ll take a look

The shape evaluation transfer should do what you want in the mean time

**Reply by GiudGiud** [reply]
It should already work
see the tests here:
test/tests/transfers/multiapp_copy_transfer/aux_to_primary

**Reply by heinono1** [reply]
Thanks. I'll take a look again, but it will take a few days.

**Reply by GiudGiud** [reply]
Sounds good. Feel free to paste input files here if you can share them
